src/Microsoft.DotNet.Interactive/ISupportSetValues.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Interactive
+{
+    public interface ISupportSetValue
+    {
+        public Task SetValueAsync(string name, object value, Type declaredType = null);
```

Comment:
Do we actually care deeply about declared vs actual type here?

Reply:
if declared type is passed we use that as type of the variable instead of the actual type, that helps when we want to ensure we use an explicit type for that value declaration